adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -251,21 +253,38 @@ def _get_relations_from_condition(
                     raise RuntimeError(
                         "Don't yet handle multiple fillers for the same semantic role"
                     )
-            first_relation_slot_filler_role_in_action = only(
-                entities_to_roles[condition.first_slot]
-            )
-            situation_object_1 = only(
-                action_roles_to_fillers[first_relation_slot_filler_role_in_action]
-            )
-            second_relation_slot_filler_role_in_action = only(
-                entities_to_roles[condition.second_slot]
-            )
-            situation_object_2 = only(
-                action_roles_to_fillers[second_relation_slot_filler_role_in_action]
-            )
 
-            perception_1 = self._objects_to_perceptions[situation_object_1]
-            perception_2 = self._objects_to_perceptions[situation_object_2]
+            # Check for manipulator in first slot
+            if CAN_MANIPULATE_OBJECTS in condition.first_slot.properties:
```

Comment:
I think we want to check first whether the slot corresponds to one of the known argument role fillers. If it does, we should use it. We should only go searching for the manipulator if the binding isn't given explicitly.